### DISCUSSION TITLE
Blazor PWA not fetching app when its cached -> cannot work offline

### DISCUSSION THREAD

**Discussion by Laftek** [dotnet/aspnetcore#54830] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I converted existing Blazor WASM app to PWA and deployed as static page to cloudflare pages. On the first visit it works fine but every other visit with app already cached I get this in console
`The FetchEvent for "https://incomecalculator.pages.dev/" resulted in a network error response: a redirected response was used for a request whose redirect mode is not "follow". Promise.then (async) (anonymous)	@	service-worker.js:7`

Repo:
https://github.com/Laftek/IncomeCalculator

App:
https://[incomecalculator.pages.dev](https://incomecalculator.pages.dev/)/

My main focus is to have this app working offline. Thank you.

### Expected Behavior

To work even on repeat visit and most important to work offline.

### Steps To Reproduce

Deploy the PWA app to cloudflare static pages
Visit the page after app is cached

### .NET Version

8.0

**Comment by mkArtakMSFT** [comment]
Hi. It looks like this is a question about how to use ASP.NET Core. While we do our best to look through all the issues filed here, we are not a general-purpose forum. To get a faster response we suggest posting your questions to [StackOverflow](https://www.stackoverflow.com) using the **blazor** tag.

**Comment by danroth27** [comment]
@Laftek Do you know what the source of the redirect is for the initial request?

**Reply by Laftek** [reply, marked as answer]
Hi @danroth27 , thanks for reaching out.  I am not sure what you mean completely but I was able to fix it thanks to one guy here on github (cannot find that issue again to give him a credit) but basically the problem is with catching so I just added few lines to service worker to give new response if its redirected.

PS:
found the guy [andreimilto](https://github.com/andreimilto) -> https://github.com/dotnet/aspnetcore/issues/33872 Thanks!
